### PR TITLE
Feature/subscribers revamping

### PR DESCRIPTION
**Pull request by aarushXL**

### CODE REVIEW

File: src/controllers/medicalFiles.controller.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    const getSurgeryHistoriesController = async (req, res) => {
+        try {
+            let findSurgeryHistoriesQuery = {};
+            const {limit, searchQuery , skip} = req.query;
+            
+            if (searchQuery) {
+                let searchKeyText = new RegExp(searchQuery, 'i');
+                findSurgeryHistoriesQuery['$or'] = [{name : searchKeyText}, {id : searchKeyText}]
+            } 
+            const totalRecords = await surgeryHistory.countDocuments(findSurgeryHistoriesQuery);
```
Comment (afourteia) [inline src/controllers/medicalFiles.controller.js:88]: Use the Mongoose function for count in a similar fashion to they way it is done in subscriber service

File: src/schemas/subscriberSchema.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // importing mongoose dependency for subscriber schema and model creation
 const mongoose = require(`mongoose`);
-const audit = require(`./auditSchema`);
+// const audit = require(`./auditSchema`);
 
 // medicalFile schema or structure
-const medicalFileSchema = mongoose.Schema({
+const medicalFileSchema = new mongoose.Schema({
```
Comment (afourteia) [inline src/schemas/subscriberSchema.js:6]: I think this can be removed.

**Review by afourteia — CHANGES_REQUESTED** [review]

**Comment by aarushXL** [comment]
I have fixed the changes as you suggested

File: src/helpers/medicalFilesHelper.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+exports.findDocuments = async (schema ,findQuery, skip, limit) => {
```
[the remark this answers is not in the document]
Comment (aarushXL) [reply src/helpers/medicalFilesHelper.js]: followed the above convention

**Review by afourteia — APPROVED** [review]

**Pull request merged**